### ISSUE TITLE
[Nightly] Salvage System broken

### ISSUE THREAD

**Issue by kernhe** [MegaMek/mekhq#4219]
# Environment
V: 0.49.20 Snapshot
O: linux (Ubuntu 20.04.4)
J: openjdk 11.0.23+9-1ubuntu1~20.04.2
Window Manager: Xfwm4 4.14
Desktop Environment: Xfce 4.14

# Description
This bug is a bit weird as it has multiple implications and it is more than the AutoAwards to blame...

## Salvage after Battle
When you win a battle and you come to the salvage screen, you see that all salvageable units have a cost of 0 (zero). See screenshot:
![ZeroCostSalvage](https://github.com/MegaMek/mekhq/assets/1056746/46419199-e39d-445a-b55f-1f17595463f0)

## Take over your salvage
MekHQ states correctly that a new unit has been added
![ZeroCostSalvage2](https://github.com/MegaMek/mekhq/assets/1056746/9a52b363-dcfd-4272-8f73-54ef8e10cc90)

## Salvaged unit becomes a Ghost unit
The salvaged unit is not in the hangar and when I want to advance a day, MekHQ complains that there are unmaintained units. But the unmaintained list is empty, as you can see in the next screenshot
![ZeroCostSalvage3](https://github.com/MegaMek/mekhq/assets/1056746/fd796431-d99c-4fa4-a284-237a05617742)

# Replicate the Bug:
1. load the Beta Testers Campaign (see files)
2. Go to Briefing Room, click Scenario "MekBase - Hostile - Recon"
3. Load Saved Game (see files)
4. Ignore the battle, just type /victory
5. see salvage dialog with 0 Credits

### Files
I just started a Beta Tester Force with vanilla units, so there is no need to share Customs.
* [BetaTesters30350329.cpnx.gz](https://github.com/user-attachments/files/15848378/BetaTesters30350329.cpnx.gz)
* Dummy battle to be loaded in the battle up in the Briefing Room: 
[autosave_2024-06-15_17-16-17.sav.gz](https://github.com/user-attachments/files/15848382/autosave_2024-06-15_17-16-17.sav.gz)

**Comment by IllianiCBT** [comment]
And, to confirm, this isn’t the Finance Tab bug?

**Comment by repligator** [comment]
![Screenshot_20240615_120841](https://github.com/MegaMek/mekhq/assets/114337591/1810c290-a6c1-4ba9-a478-5721019e164c)
Does the finance tab bug zero out everything or just somethings?

**Comment by Thom293** [comment]
Did you change the parts rating systems to swap the A-F thing?  I think there is a bug related to that.

**Comment by kernhe** [comment]
@Thom293:
a) A-F thing: negative, sir! ... I left it as it is.
b) Finance Tab bug: not aware of this, but my Warehouse was posted by @repligator 
    
I even advanced further some months with my Beta Testers and on first sight it looks fine in the Finances tab. BUT I opt for 
* pay for parts
* pay for repairs
... and this is not working correctly. I have lots of repair entries in Finances but no Debit no Credit and no change in Balance.


Will try a new Beta Testers with AtB-rules and only make pay for parts and repairs next week.

**Comment by repligator** [comment]
Using a different I can't replicate the financial issues. I have however experienced "ghost mech" bugs; where mechs can't be seen until the save is reloaded.

**Comment by IllianiCBT** [comment]
I’m on vacation till Monday, but I’ll prioritize this when I return. I’m at a loss as to how this could be caused by autoAwards though. Does it go away when autoAwards is disabled?

**Comment by Thom293** [comment]
On the ghost thing it may take them a few days to appear in hanger.  They appear in repair bay before hanger for some reason. But if they don't and have a repeatable ghot big that would be helpful.  The 0 value I am confused on though.

**Comment by IllianiCBT** [comment]
My theory is that it’s due to the new variable quality salvage option. But I didn’t encounter this during testing, so it’s probably a combination of things.

**Comment by kernhe** [comment]
Next week I'll make more test cases, especially I will not alter the financial system besides paying for parts and repairs.

I will also check if ghosts will manifest over time. ... but I doubt this as in the past salvaged mechs are instantly available in repair bay while your deployed units will return after "min deploy time".

**Comment by kernhe** [comment]
> I’m on vacation till Monday, but I’ll prioritize this when I return. I’m at a loss as to how this could be caused by autoAwards though. Does it go away when autoAwards is disabled?

Not checked if I disable autoAward. But my guessing is based on gut's feeling and the experience made with looking through the code during searching for the kill counter bug and trying to understand how the AwardSystem was welded on the code. Looks like a Cataphract to me...

**Comment by IllianiCBT** [comment]
While it’s not outside the realms of possibility, autoAwards doesn’t interact with loot at all, nor does it interact with finances. So I’d be very surprised if that was behind this.

It’s far more likely to be the random quality code that got pushed before the weekend. As that indirectly interacts with the value of the object by changing its quality.

I’ll try and find some time to take a look once my guests have gone home :)

**Comment by IllianiCBT** [comment]
It's being caused by the Finance Tab bug:

![image](https://github.com/MegaMek/mekhq/assets/103902653/adef8587-680f-4cc3-b731-41d5443816f4)

**Comment by IllianiCBT** [comment]
So, now I've confirmed it's due to the finance tab having been reset to 0, I have a few follow-up questions:

- when was this campaign created (irl date)?
- are you using one of the pe-made presets, or a custom one (this includes any premades you might have customized)?
- have you interacted with the 'Reverse quality names' option in the 'Repair and Maintenance' tab? I can see it's currently disabled, so I'm assuming not, but I'd be remiss not to ask.

**Comment by kernhe** [comment]
@IllianiCBT to your questions:
1. 3035
2. I used a custom where that finance tab bug went in
3. nope

I do not know how that setting went in there, as I did not altered any of this values. I just made my checkmarks (pay for this and that) and went on. But now I see the Zeros.

I tried to replicate the changes and save new custom presets (three settings) but everything went fine.

TLDR:
* it is the Finance Tab bug, in combination with user made custom preset that contained that bug.

**Comment by kernhe** [comment]
I just checked my new preset file, after I fixed the usedPartPriceMultipliers and found something odd:

    <usedPartPriceMultipliers>0.1,0.2,0.3,0.5,0.7000000000000001,0.9</usedPartPriceMultipliers> 

all values were set using up-/down-arrows. But where does this precision of 0.7000000000000001 come from?!?

Maybe this might lead to the zeroing out all other values?

**Comment by IllianiCBT** [comment]
To my best knowledge that's just a rounding error and can be safely ignored.

**Comment by IllianiCBT** [comment]
Closing as resolved
